src/ui/pages/ConnectedSites/ConnectedSites.tsx

```diff
@@ -143,38 +148,120 @@ function ConnectedSitesList({
   );
 }
 
+export function ConnectedSitesSearch({
+  value,
+  onChange,
+}: {
+  value?: string;
+  onChange(value: string): void;
+}) {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const debouncedHandleChange = useDebouncedCallback(onChange, 300);
+
+  useLayoutEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.value = value || '';
+    }
+  }, [value]);
+
+  return (
+    <SearchInput
+      ref={inputRef}
+      boxHeight={40}
+      type="search"
+      placeholder="Search"
+      defaultValue={value}
```

Comment:
Can we make this input fully controllable?

Reply:
Refactored, please check out the updated version ✅